src/MembershipContextFactory.php

```diff
@@ -4,97 +4,73 @@
 
 namespace WMDE\Fundraising\MembershipContext;
 
-use Doctrine\DBAL\Connection;
-use Doctrine\DBAL\DriverManager;
-use Doctrine\ORM\EntityManager;
-use Pimple\Container;
-use Pimple\ServiceProviderInterface;
-use WMDE\Fundraising\MembershipContext\Authorization\ApplicationAuthorizer;
+use Doctrine\Common\Annotations\AnnotationReader;
+use Doctrine\Common\EventSubscriber;
+use Doctrine\Common\Persistence\Mapping\Driver\MappingDriver;
+use Doctrine\ORM\Configuration;
+use Gedmo\Timestampable\TimestampableListener;
 use WMDE\Fundraising\MembershipContext\Authorization\MembershipTokenGenerator;
 use WMDE\Fundraising\MembershipContext\Authorization\RandomMembershipTokenGenerator;
-use WMDE\Fundraising\MembershipContext\DataAccess\DoctrineApplicationAuthorizer;
 use WMDE\Fundraising\MembershipContext\DataAccess\DoctrineMembershipApplicationPrePersistSubscriber;
-use WMDE\Fundraising\Store\Factory as StoreFactory;
-use WMDE\Fundraising\Store\Installer;
 
 /**
  * @licence GNU GPL v2+
  * @author Jeroen De Dauw < jeroendedauw@gmail.com >
  */
-class MembershipContextFactory implements ServiceProviderInterface {
-
-	private $config;
+class MembershipContextFactory {
 
 	/**
-	 * @var Container
+	 * Use this constant for MappingDriverChain::addDriver
 	 */
-	private $pimple;
+	public const DOCTRINE_ADDITIONAL_ENTITIES = [
```

Comment:
We don't need several namespaces, let's do the same as I did on donations, make this a string and call it `ENTITY_NAMESPACE`. We don't need to expose the file paths to the outside, that's only for the mapping driver.